libkbfs/block_retrieval_queue.go

```diff
@@ -173,7 +173,7 @@ func (brq *blockRetrievalQueue) Request(ctx context.Context, priority int, kmd K
 			// specific type where the request blocks are CommonBlocks, but
 			// that direction can Set correctly. The cache will never have
 			// CommonBlocks.
-			cachedBlock, hasPrefetched, err :=
```

Comment:
The blocks re put in the transient cache in `folderBranchOps.finalizeBlocks`, after the MD is successfully put to the mdserver.

And I don't think that race can happen here, since `Request()` should never see permanent entries.  I think it also can't happen in `getBlockHelperLocked` either, since `blockLock` is taken across both the `GetWithPrefetch` and `PrefetchAfterBlockRetrieved` calls, as well as the `DeletePermanent` call.